### ISSUE TITLE
formula.api  cleanup remove classes

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#833]
I think it would be better to delete the non-formula classes from formula.api

```
>>> import statsmodels.formula.api as smf
>>> dir(smf)
['GLM', 'GLS', 'GLSAR', 'Logit', 'MNLogit', 'NegativeBinomial', 'OLS', 'Poisson', 
'Probit', 'RLM', 'WLS', '__builtins__', '__doc__', '__file__', '__name__', 
'__package__', 'glm', 'gls', 'glsar', 'logit', 'mnlogit', 'negativebinomial', 
'ols', 'poisson', 'probit', 'rlm', 'wls']
```

**Comment by josef-pkt** [comment]
bump

I just needed the debugger to figure out that I had use smf.NegativeBinomial instead of smf.negativebinomial

**Comment by jbrockmendel** [comment]
Closed by #5146.
